pkg/agent/controller/networkpolicy/cache.go

```diff
@@ -379,6 +388,31 @@ func (c *ruleCache) processPodUpdate(e interface{}) {
 	}
 }
 
+// processExternalNodeUpdate will be called when ExternalNodeController publishes an ExternalNode update event.
+// It finds out AppliedToGroups that contains this ExternalNode converted ExternalEntity and trigger reconciling
+// of related rules.
+// It can enforce NetworkPolicies to ExternalEntities after ExternalEntityInterface is realised in the interface store.
+func (c *ruleCache) processExternalNodeUpdate(e interface{}) {
+	externalNode := e.(*crdv1alpha1.ExternalNode)
+	eeName := externalnode.GenExternalEntityName(externalNode)
+	c.appliedToSetLock.RLock()
+	defer c.appliedToSetLock.RUnlock()
+	externalEntityEquals := func(ee *v1beta.ExternalEntityReference, namespace string, name string) bool {
+		if ee.Namespace == namespace && ee.Name == name {
+			return true
+		}
+		return false
+	}
+	for group, memberSet := range c.appliedToSetByGroup {
+		for _, member := range memberSet.Items() {
```

Comment:
can it use `memberSet.Has` like `processPodUpdate`? will members in `c.appliedToSetByGroup` have IPs?

Reply:
Thanks for the suggestion, let me check this part further to see if memberSet.Has can work. Do you mean if interface contains IPs in this case? Yes, ExternalNodeController will notify NetworkPolicyController after the ExternalNode is created or updated and the IPs exist for this interface in the interfaceStore.

Reply:
If the member in `c.appliedToSetByGroup` has IPs set, `memberSet.Has` cannot be used to query because when members are stored, their keys are populated based on name+namespace+ips, but when we query it, we don't have IPs, so keys will be different.

Reply:
Thanks for the information! For ExternalEntity, controller will populate IPs for group member https://github.com/antrea-io/antrea/blob/main/pkg/controller/networkpolicy/networkpolicy_controller.go#L1197.
I think that name and namespace is enough for ExternalEntity case, so I revert the code to the first version.

Reply:
Have updated by notifying ExternalEntityRef.